src/main/java/com/google/devtools/build/lib/actions/PathMapper.java

```diff
@@ -105,6 +105,16 @@ default String getMappedExecPathString(ActionInput artifact) {
     return map(artifact.getExecPath()).getPathString();
   }
 
+  /**
+   * Returns the length of the mapped path minus the length of the unmapped path.
+   *
+   * <p>Implementations should provide a more efficient implementation that avoids allocations.
+   */
+  default int computeExecPathLengthDiff(Artifact.DerivedArtifact artifact) {
```

Comment:
Import `DerivedArtifact` instead of qualifying it. A qualifier is not needed if the inner type is unambiguous.